src/ocl/convolutionocl.cpp

```diff
@@ -1591,6 +1540,164 @@ static inline bool IsAlgorithmDisabled(const miopenConvAlgorithm_t algo)
     } // clang-format on
 }
 
+// Helper class used for emplace and sort.
+struct SolutionSortWrapper : miopenConvSolution_t
+{
+    SolutionSortWrapper(const float& t,
+                        const size_t& ws,
+                        const uint64_t& id,
+                        const miopenConvAlgorithm_t& algo)
+        : miopenConvSolution_t{t, ws, id, algo}
+    {
+    }
+    bool operator<(const SolutionSortWrapper& other) const { return (time < other.time); }
+};
+
+void ConvolutionDescriptor::GetSolutionsFallback(Handle& handle,
+                                                 const ProblemDescription& problem,
+                                                 const size_t maxSolutionCount,
+                                                 size_t* const solutionCount,
+                                                 miopenConvSolution_t* const solutions) const
+{
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMMED_FALLBACK{}))
+    {
+        MIOPEN_LOG_I("Disabled via environment");
+        *solutionCount = 0;
+        return;
+    }
+
+    /// \todo This is terrible. Should do away when we converge to
+    /// single conv::ProblemDescription type.
+    const auto& inDesc = problem.direction.IsForward() ? problem.conv_problem.GetIn()
+                                                       : problem.conv_problem.GetOut();
+    const auto& outDesc = problem.direction.IsForward() ? problem.conv_problem.GetOut()
+                                                        : problem.conv_problem.GetIn();
+    const auto& weightsDesc = problem.conv_problem.GetWeights();
+    // This check is needed on fallback path only.
+    // On regular path (find-db hit) this was checked during Find().
+    ValidateGroupCount(inDesc, weightsDesc, *this);
+
+    std::vector<SolutionSortWrapper> interim;
+    interim.reserve(maxSolutionCount); // For speed. In most cases we have less entries than asked.
+
+    auto ctx = ConvolutionContext{problem};
+    ctx.SetStream(&handle);
+    ctx.DetectRocm();
+
+    const auto& map = miopen::solver::GetMapValueToAnySolver();
+    for(const auto& item : map)
+    {
+        const auto solver_id = solver::Id{item.first};
+        // solver_id is always valid here, because taken from registry.
+        // Validity check is not required.
+        const auto algo = solver_id.GetAlgo();
+        if(IsAlgorithmDisabled(algo)) // Algos can be disabled globally.
+            continue;
+        const auto& s = item.second;
+        if(!s.IsApplicable(ctx))
+            continue;
+        if(!s.IsDynamic()) // Let's allow non-dynamic later, if necessary.
+        {
+            MIOPEN_LOG_I2(solver_id.ToString() << " Not dynamic, skipped");
+            continue;
+        }
```

Comment:
Can we check for dynamic first and `isApplicable` later. In some cases `IsApplicable` has a considerable cost

Reply:
Fixed!